src/main/java/at/laborg/briss/gui/DrawableCropRect.java

```diff
@@ -50,37 +63,86 @@ public DrawableCropRect(int x, int y, int width, int height) {
     }
 
     private void setupCorners() {
-        cornerUpperLeft = new Rectangle(CORNER_DIMENSION, CORNER_DIMENSION);
-        cornerUpperRight = new Rectangle(CORNER_DIMENSION, CORNER_DIMENSION);
-        cornerLowerLeft = new Rectangle(CORNER_DIMENSION, CORNER_DIMENSION);
-        cornerLowerRight = new Rectangle(CORNER_DIMENSION, CORNER_DIMENSION);
+        cornerUpperLeft = new ResizeHandle();
+        cornerUpperRight = new ResizeHandle();
+        cornerLowerLeft = new ResizeHandle();
+        cornerLowerRight = new ResizeHandle();
     }
 
     private void setCornerLocations(int x, int y) {
-        int halfCornerDimension = CORNER_DIMENSION / 2;
-        cornerUpperLeft.setLocation(x - halfCornerDimension, y - halfCornerDimension);
-        cornerUpperRight.setLocation(x + width - halfCornerDimension, y - halfCornerDimension);
-        cornerLowerLeft.setLocation(x - halfCornerDimension, y + height - halfCornerDimension);
-        cornerLowerRight.setLocation(x + width - halfCornerDimension, y + height - halfCornerDimension);
+        cornerUpperLeft.setLocation(x, y);
+        cornerUpperRight.setLocation(x + width, y);
+        cornerLowerLeft.setLocation(x, y + height);
+        cornerLowerRight.setLocation(x + width, y + height);
     }
 
-    public void draw(Graphics2D g2) {
+    public boolean hasEnoughSpaceForHandles() {
+        return !(getWidth() < 2 * DrawableCropRect.CORNER_DIMENSION)
+            && !(getHeight() < 2 * DrawableCropRect.CORNER_DIMENSION);
+    }
+
+    private Font scaleFont(String text, FontMetrics fontMetrics) {
+
+        int size = MergedPanel.BASE_FONT.getSize();
+        float width = fontMetrics.stringWidth(text);
+        float height = fontMetrics.getHeight();
+        if (width == 0 || height == 0)
+            return MergedPanel.BASE_FONT;
+        float scaleFactorWidth = (float) (this.width * 0.95 / width);
```

Comment:
Replace magic constant 0.95 with named constant to reveal intent.

Reply:
Also, using `0.95f` would eliminate the need for the cast and improve readability.